packages/cli/__tests__/daemon/__integration__/cli.zowe.daemon.integration.suite.test.ts

```diff
@@ -0,0 +1,97 @@
+
+describe("Zowe daemon suite", () => {
+    const rimrafSync = require("rimraf").sync;
+
+    let prebuildsDir: string;
+    let zoweExePath: string;
+    let zoweExeDirPath: string;
+    let zoweExeTgzPath: string;
+    let zoweExeFileNm: string;
+
+    // establish path names and record whether we are running our EXE or node.js script
+    beforeAll(async () => {
+        // determine executable file name and TGZ path for our current OS
+        const zoweRootDir: string = nodeJsPath.normalize(__dirname + "/../../../../..");
+        prebuildsDir = nodeJsPath.normalize(zoweRootDir + "/packages/cli/prebuilds");
+        const sysInfo: ISystemInfo = ProcessUtils.getBasicSystemInfo();
+        switch (sysInfo.platform) {
+            case "darwin": {
+                zoweExeFileNm = "zowe";
+                zoweExeTgzPath = nodeJsPath.resolve(prebuildsDir, "zowe-macos.tgz");
+                break;
+            }
+            case "linux": {
+                zoweExeFileNm = "zowe";
+                zoweExeTgzPath = nodeJsPath.resolve(prebuildsDir, "zowe-linux.tgz");
+                break;
+            }
+            case "win32": {
+                zoweExeFileNm = "zowe.exe";
+                zoweExeTgzPath = nodeJsPath.resolve(prebuildsDir, "zowe-windows.tgz");
+                break;
+            }
+            default: {
+                zoweExeFileNm = "exeForUnknownOs";
+                zoweExeTgzPath += "unknownOs.tgz";
+                throw "cli.zowe.daemon.integration.suite.test.ts: beforeAll: " +
+                    sysInfo.platform + " is not a known OS.";
+            }
+        }
+
+        // Form the path name to our executable.
+        zoweExeDirPath = nodeJsPath.normalize(zoweRootDir + "/zowex/target/release");
+        zoweExePath = nodeJsPath.resolve(zoweExeDirPath, zoweExeFileNm);
+        if (!IO.existsSync(zoweExePath)) {
+            zoweExeDirPath = nodeJsPath.normalize(zoweRootDir + "/zowex/target/debug");
+            zoweExePath = nodeJsPath.resolve(zoweExeDirPath, zoweExeFileNm);
+            if (!IO.existsSync(zoweExePath)) {
+                zoweExePath = "./NoZoweExeExists";
+            }
+        }
+
+        // Remove any existing prebuilds directory
+        rimrafSync(prebuildsDir);
+
+        // create our prebuilds directory
+        IO.createDirSync(prebuildsDir);
+
+        // tar our EXE into a TGZ for this platform, so that we can extract it in tests
+        tar.create({
+            sync: true,
+            gzip: true,
+            cwd: zoweExeDirPath,
+            file: zoweExeTgzPath
+        }, [zoweExeFileNm]);
+    });
+
+    afterAll(async () => {
+        // Remove any existing prebuilds directory
+        rimrafSync(prebuildsDir);
+    });
+
+    // run all of our daemon-related tests
+    require("./cli.zowe.exe.integration");
+    require("./enable/cli.daemon.enable.integration");
+    require("./disable/cli.daemon.disable.integration");
```

Comment:
This is a clever way to combine multiple test suites into one 🙂 I'm curious if there are any downsides of having test files where the filename doesn't end in `.test.ts`. None come to mind immediately, since our `checkTestsCompile` and `lint:test` scripts process all files that are in `__tests__` directories.

Reply:
I considered that checkTestsCompile should confirm that the xxx.non-test.ts files compile, which is desireable. I purposely wanted the required files to not be xxx.test.ts, since jest runs all xxx.test.files. That would mean that the tests run twice, once from the xxx.suite.test.ts file and once from themselves. Some of the individual test files would fail because they would not get some of the setup done in the suite.

Downsides:
- I had to duplicate some logic in the suite and again in the individual tests, because I did not want variables declared in the suite to be used within the individual test files without being declared in the individual test file. It would work, but it would be very confusing.
- You may not be able to run some of the individual test files directly with jest. The suite creates the tgz, so if an individual test file needs the tgz, that test will fail.
    - A decent workaround is that you can always run the suite file, but still specifiy the text of an "it" or "describe" statement to limit the test to one or a few tests.